src/test/java/org/reaktivity/nukleus/maven/plugin/internal/Varbyteint32FW.java

```diff
@@ -0,0 +1,148 @@
+package org.reaktivity.nukleus.maven.plugin.internal;
+
+import org.agrona.DirectBuffer;
+import org.agrona.MutableDirectBuffer;
+import org.agrona.concurrent.UnsafeBuffer;
+import org.reaktivity.reaktor.internal.test.types.Flyweight;
+
+import static java.lang.Byte.MAX_VALUE;
+
+public class Varbyteint32FW extends Flyweight {
+    private final static int BYTE_MASK = 0xFF;
+
+    private final static int CONTINUATION_BIT = 0x80;
+
+    private final static int MAX_INPUT = 0x0FFFFFFF;
+
+    private final static int MAX_MULTIPLIER = 0x200000;
+
+    private int size;
+
+    public static void main(String[] args) {
+        Varbyteint32FW.Builder varbyteint32RW = new Varbyteint32FW.Builder();
+        MutableDirectBuffer writeBuffer = new UnsafeBuffer(new byte[8192]);
+
+        Varbyteint32FW varbyteint32 = varbyteint32RW.wrap(writeBuffer, 0, writeBuffer.capacity())
+                .set(268435455)
+                .build();
+
+        final int value = varbyteint32.value();
+        System.out.printf("value=%d, %d\n", value, varbyteint32.length0());
+    }
+
+    @Override
+    public int limit() {
+        return offset() + size;
+    }
+
+    public int value() {
+        int value = 0;
+        int multiplier = 1;
+        int pos  = offset();
+        int encodedByte;
+        do {
+            encodedByte = buffer().getByte(pos++);
+            value += (encodedByte & MAX_VALUE) * multiplier;
+            if (multiplier > MAX_MULTIPLIER) {
+                throw new IllegalArgumentException("varbyteint32 value is too long");
+            }
+            multiplier *= CONTINUATION_BIT;
+        }
+        while ((encodedByte & CONTINUATION_BIT) != 0);
+        return value;
+    }
```

Comment:
```
public int value()
{
    final DirectBuffer buffer = buffer();
    final int offset = offset();
    final int limit = limit();

    int value = 0;
    int index = 0;
    byte b;
    do
    {
        b = buffer.getByte(offset + index);
        value <<= 7;
        value |= b & ~CONTINUATION_BIT;
        index++;
    } while (offset + index < limit && (b & CONTINUATION_BIT) != 0)

    return value;
}
```

Calling `value()` assumes that `wrap` or `tryWrap` was successful, so no need to re-enforce checks.